src/stores/Frame/Region/RegionSetStore.ts

```diff
@@ -71,6 +71,28 @@ export class RegionSetStore {
         return regionId;
     };
 
+    /**
+     * Returns an array of region IDs and types in the region set.
+     */
+    @computed get regionList(): {id: number; type: CARTA.RegionType}[] {
+        return this.regions.map(r => {
+            return {id: r.regionId, type: r.regionType};
+        });
+    }
+
+    /**
+     * Returns a map associating region IDs with their corresponding RegionStore instances.
+     */
+    @computed get regionMap(): Map<number, RegionStore> {
+        const reigonMap = new Map<number, RegionStore>();
+
+        for (const region of this.regions) {
+            reigonMap.set(region.regionId, region);
+        }
+
+        return reigonMap;
```

Comment:
Minor issue: there's a typo in the local variable here.

Reply:
Fixed.